### PR TITLE
Don't load yaml booleans as python booleans

### PR DESCRIPTION
**Pull request by redhatrises**
#### Description:

- Don't load yaml booleans as python booleans

#### Rationale:

- Booleans were being loaded as `True`, `False`, etc.

### CODE REVIEW

**Comment by matejak** [comment]
What is the reason for this change? Do we have `true`/`false` values in a non-boolean context? I am uncomfortable with silent mass conversion of any passed boolean value to a string.

**Comment by matejak** [comment]
@openscap-ci test this please

**Comment by redhatrises** [comment]
@openscap-ci test this please

**Comment by matejak** [comment]
I don't think that we have to hack the yaml format parsing - those `true`, `on` etc. values [more on those here](http://yaml.org/type/bool.html) can be protected by double quotes, so `"true"` will be parsed as strings.
I think that raising an exception when an input value is expected to be string, but it is boolean would lead to a more strict requirements on input, which could be beneficial for the project.

**Comment by redhatrises** [comment]
@matejak I would like to not have to have the quotes if possible and not to have that nuance that could be confusing to some people. It would also make it easier ignore case for whatever type of boolean and not have to remember to have quotes.

**Comment by matejak** [comment]
OK, let's collect opinions of more people than just two of us, this is quite a substantial change.

**Comment by redhatrises** [comment]
Agreed. I updated the code to handle any case of booleans e.g. `true`, `True`, `false`, `on`, `Off`, etc.

File: shared/utils/yaml-to-shorthand.py [hunk 1]
```diff
@@ -10,13 +10,29 @@
...
+def construct_yaml_bool(self, node):
+    value = self.construct_scalar(node)
+
+    return value
+
+
+def bool_constructor(self, node):
+    value = construct_yaml_bool(self, node)
+
+    return value
```
Comment (mpreisler) [inline shared/utils/yaml-to-shorthand.py]: this is python code, not a Tolkien novel :-P Please do this on just 2 lines.

Comment (redhatrises) [reply shared/utils/yaml-to-shorthand.py]: But @mpreisler I like Tolkien novels... maybe that is why it is written as such. :D

Comment (redhatrises) [reply shared/utils/yaml-to-shorthand.py]: @mpreisler Tolkien novel reduced to Ernest Hemingway's shortest novel.

**Comment by dahaic** [comment]
@redhatrises, I agree with @matejak. If we say we use yaml, we should stick to the specification. It might be less user friendly, but then we can just recommend to quote everything and be done with it.

**Comment by redhatrises** [comment]
Remember that we are doing a YAML-like conversion, so we don't actually have to stick 100% with yaml standards. In fact, we may not fully be able to. Plus, this is going to be documented in the developer guide. Remember the easier that we can make this, without having all the nuances, the better for everyone.

**Comment by dahaic** [comment]
I have changed my mind. This YAML "feature" is quite obscure anyway, and has probably zero usage in SSG. I am for accepting this :)

**Comment by shawndwells** [comment]
I don't care as long as it's documented somewhere

**Comment by dahaic** [comment]
Thanks @matejak for challenging the idea :) Good stuff, @redhatrises !! Ack.

**Comment by redhatrises** [comment]
Yes. Thanks @matejak for challenging the idea as well.

**Pull request merged**